### PR TITLE
Documentation: update pot files of docs for translation and upload to transifex

### PR DESCRIPTION
**Pull request by unkcpz**
Currently I am placing this process in the deploy stage of the CI.
Because the large area update of the documentation is usually accompanied by a new tag.
I wonder if this is reasonable, will it disturb the original deploy stage?

The availability of the CI has been tested and passed in my forked repository.
You need to refer to https://docs.transifex.com/integrations/github#integrating-the-client-with-travis-ci to add environment variables TRANSIFEX_USER and TRANSIFEX_PASSWORD to travis.

### CODE REVIEW

**Comment by ltalirz** [comment]
Thanks a lot @unkcpz !

> Currently I am placing this process in the deploy stage of the CI. Because the large area update of the documentation is usually accompanied by a new tag. 

I think this is a very sensible choice.

> I wonder if this is reasonable, will it disturb the original deploy stage?

Hm... probably not. Still, I would prefer to add it as a separate job.

> The availability of the CI has been tested and passed in my forked repository.

You mean CI in general or also the deploy stage? Would you mind triggering the deploy stage on your fork by adding a tag and then linking the output of travis here?

> You need to refer to https://docs.transifex.com/integrations/github#integrating-the-client-with-travis-ci to add environment variables TRANSIFEX_USER and TRANSIFEX_PASSWORD to travis.

Done!

File: .ci/transifex_upload.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+set -ev
+
+
+pip install virtualenv
+virtualenv ~/env
+source ~/env/bin/activate
+pip install transifex-client sphinx-intl
+pip install ".[docs,testing]"
+
+# Create the main database
```
Comment (ltalirz) [inline .ci/transifex_upload.sh]: Is this is necessary? In principle [it should not be necessary to have a DB for compiling the documentation](https://github.com/aiidateam/aiida_core/blob/6cd8c6ef4ce01ccf864b0c3a60d7d4d7cb0e9d44/docs/source/conf.py#L28)
If there are problems you could also play with setting the `READTHEDOCS` environment variable to `TRUE`, see [here](https://github.com/aiidateam/aiida_core/blob/6cd8c6ef4ce01ccf864b0c3a60d7d4d7cb0e9d44/docs/source/conf.py#L243) and following lines.

File: .travis.yml [hunk 2]
```diff
@@ -83,8 +83,14 @@ jobs:
     include:
         - stage: deploy
           if: "tag =~ ^v[0-9]+\\.[0-9]+\\.[0-9]+(a|b|rc)?[0-9]*$"
-          services: ignore
-          addons: skip
+          services: postgresql
```
Comment (ltalirz) [inline .travis.yml]: again, on readthedocs this is also not available so it should be possible to do this without a database

**Comment by ltalirz** [comment]
In the meanwhile, I will use your code to upload the first version of the .pot file to transifex :-)

File: .ci/transifex_upload.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Set the main profile as the default
+verdi profile setdefault $TEST_AIIDA_BACKEND
+
+# Set the polling interval to 0 otherwise the tests take too long
+verdi config runner.poll.interval 0
+
+# Generate pot file and upload to transifex platform
+TRANSIFEX_PROJECT_NAME="aiida-core"
+sphinx-build -b gettext docs/source locale
+tx init --no-interactive
```
Comment (ltalirz) [inline .ci/transifex_upload.sh]: You mentioned that the API documentation should be excluded.

The way to achieve this (tested locally) is to add the following lines [here](https://github.com/aiidateam/aiida_core/blob/6cd8c6ef4ce01ccf864b0c3a60d7d4d7cb0e9d44/docs/source/conf.py#L304):
```python
 def setup(app):
-    app.connect('builder-inited', run_apidoc)
+    # Build of API documentation may be disabled
+    # (used for translation for transifex)
+    if os.environ.get('RUN_APIDOC', None) != 'False':
+        app.connect('builder-inited', run_apidoc)
```
and to `export RUN_APIDOC=False`

**Comment by unkcpz** [comment]
Sorry for recommit and git push. I trigger this accidentally.
I am not familiar with the Github workflow for updating a PR.
Do I need to cleaning commit history or squash commit when I update the code?
@ltalirz

**Comment by ltalirz** [comment]
> Do I need to cleaning commit history or squash commit when I update the code?
@ltalirz

Thanks for asking. Ideally, you clean your commit history by rebasing on top of develop (`git rebase -i develop`, followed by `git push --force`), for more details see [here](https://github.com/aiidateam/aiida_core/wiki/Git-style-requirements).
This is most relevant for larger PRs, however, where you want to keep multiple separate commits.

Since your changes will fit inside one commit, we will anyhow "squash and merge" your PR into `develop`, cleaning your history.

**Comment by unkcpz** [comment]
> You mean CI in general or also the deploy stage? Would you mind triggering the deploy stage on your fork by adding a tag and then linking the output of travis here?

As your advice, I put it in a separate stage and trigger it by setting `if: branch=<branch-name>`. The link to the CI result: https://travis-ci.org/unkcpz/aiida_core/jobs/534661501

And remove the DB dependent in CI by setting `READTHEDOCS=True` and `RUNAPI=False`.

**Pull request merged**